app/src/main/java/com/example/android/yourcity/data/repository/CountryRepository.java

```diff
@@ -1,58 +1,124 @@
 package com.example.android.yourcity.data.repository;
 
-import android.util.Log;
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
 
-import com.example.android.yourcity.App;
-import com.example.android.yourcity.data.model.Country;
+import com.example.android.yourcity.data.model.xml.Entry;
+import com.example.android.yourcity.data.remote.Api;
+import com.example.android.yourcity.data.remote.ApiXML;
+import com.example.android.yourcity.ui.home.CallbackCountry;
 import com.google.gson.Gson;
-import com.google.gson.JsonObject;
 
 import org.json.JSONArray;
 import org.json.JSONException;
 import org.json.JSONObject;
 
-import java.net.MalformedURLException;
-import java.net.URL;
 import java.util.ArrayList;
-import java.util.Iterator;
 import java.util.List;
 
+import javax.inject.Inject;
+
 import retrofit2.Call;
 import retrofit2.Callback;
 import retrofit2.Response;
-import retrofit2.http.Url;
 
 public class CountryRepository {
 
+    private final Context context;
+    private final Api api;
+    private final ApiXML apiXML;
     private List<String> countries;
+    private List<String> cities;
+    private String cityDescription;
 
-    public CountryRepository() {
+    @Inject
+    public CountryRepository(Context context, Api api, ApiXML apiXML) {
+        this.api = api;
+        this.apiXML = apiXML;
         countries = new ArrayList<>();
+        cities = new ArrayList<>();
```

Comment:
If you use 'this ' in the constructor, use it everywhere please
e.g this.countries

Reply:
Make lists final too